### ISSUE TITLE
Error in console

### ISSUE THREAD

**Issue by Weimmer** [keijiro/AIShader#1]
Hi Keijiro,

it is a very cool test, that's why I grab as soon as I see it, but when I click the generate button on the AIShader, it has error in the console:

NullReferenceException: Object reference not set to an instance of an object
AIShader.AIShaderImporterEditor.Regenerate () (at Assets/Editor/AIShaderImporterEditor.cs:55)
`_cached.stringValue = response.choices[0].message.content;`

Maybe I missed some setting? I have already pasted my chatgpt api key, where it should to be.

Best regards
Weimmer

**Comment by keijiro** [comment]
Could you check if your API key works properly for the chat completion API? You can test it with curl like the following:

```
curl https://api.openai.com/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer YOUR_API_KEY' \
  -d '{
  "model": "gpt-3.5-turbo",
  "messages": [{"role": "user", "content": "Hello!"}]
}'
```

**Comment by Da-bass-man** [comment]
I'm having the same issue and i tested the curl you got here and got just got errors 
where would i need to run this i would assume cmd correct?

**Comment by keijiro** [comment]
@Da-bass-man Please copy-paste the errors.

(I also recommend your billing status if your free trial period has been expired.)

**Comment by Da-bass-man** [comment]
Ahh you know that might be it looks like my credits expired

**Comment by Weimmer** [comment]
> curl https://api.openai.com/v1/chat/completions \
>   -H 'Content-Type: application/json' \
>   -H 'Authorization: Bearer YOUR_API_KEY' \
>   -d '{
>   "model": "gpt-3.5-turbo",
>   "messages": [{"role": "user", "content": "Hello!"}]
> }'

Thanks for reply, I have tried and done some research, it seems I need a ChatGPT plus, is that true?

**Comment by keijiro** [comment]
No. You only need ChatGPT API access.

https://openai.com/blog/introducing-chatgpt-and-whisper-apis

**Comment by Weimmer** [comment]
> No. You only need ChatGPT API access.
> 
> https://openai.com/blog/introducing-chatgpt-and-whisper-apis

I think I got it, I have to add a payment method in my account, otherwise it will get the error like above. Thanks, it is really cool :)

**Comment by keijiro** [comment]
Yes, it needs payment settings if your free trial period has been ended.

**Comment by 501632** [comment]
> ac**xxx5***63/2 https://api.openai.com/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer YOUR_API_KEY' \
  -d '{
  "model": "gpt-3.5-turbo",
  "messages": [{"role": "", "content": "}]
}'

**Comment by 501632** [comment]
https://github.com/501632/AIShader/issues/1#issuecomment-1457760284

**Comment by 501632** [comment]
[https://github.com/501632/AIShader/issues/1#issuecomment-1457760284](https://etherscan.io/) 
Duplicate of #

**Comment by Matrixx123456** [comment]
> Hi Keijiro,
> 
> it is a very cool test, that's why I grab as soon as I see it, but when I click the generate button on the AIShader, it has error in the console:
> 
> NullReferenceException: Object reference not set to an instance of an object AIShader.AIShaderImporterEditor.Regenerate () (at Assets/Editor/AIShaderImporterEditor.cs:55) `_cached.stringValue = response.choices[0].message.content;`
> 
> Maybe I missed some setting? I have already pasted my chatgpt api key, where it should to be.
> 
> Best regards Weimmer

[](url)
